cdap-securestore-ext-gcp-secretstore/src/main/java/io/cdap/cdap/securestore/gcp/cloudsecretmanager/CloudSecretManagerClient.java

```diff
@@ -0,0 +1,244 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.securestore.gcp.cloudsecretmanager;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.rpc.ApiException;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.secretmanager.v1.AddSecretVersionRequest;
+import com.google.cloud.secretmanager.v1.CreateSecretRequest;
+import com.google.cloud.secretmanager.v1.ListSecretsRequest;
+import com.google.cloud.secretmanager.v1.Secret;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient.ListSecretsPage;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceSettings;
+import com.google.cloud.secretmanager.v1.SecretPayload;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.FieldMask;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import java.util.Optional;
+
+/** Client for <a href="https://cloud.google.com/secret-manager">Google Cloud Secret Manager</a> */
+public class CloudSecretManagerClient {
+  /**
+   * Property that, if set, overrides which GCP project to use rather than the default (see
+   * https://github.com/googleapis/google-cloud-java#specifying-a-project-id for default behavior).
+   */
+  private static final String PROJECT_ID = "project.id";
+  /**
+   * Property that, if set, overrides which service account to authorize as rather than the default
+   * (see https://github.com/googleapis/google-cloud-java#authentication for default behavior)
+   */
+  private static final String SERVICE_ACCOUNT_FILE = "service.account.file";
+
+  private final SecretManagerServiceClient secretManager;
+  // GCP project resource name e.g. "project/my-project-id".
+  private final String projectResourceName;
+
+  public CloudSecretManagerClient(Map<String, String> properties) throws IOException {
+    String projectId =
+        properties.containsKey(PROJECT_ID)
+            ? properties.get(PROJECT_ID)
+            : ServiceOptions.getDefaultProjectId();
+    this.secretManager = createClient(properties);
+    this.projectResourceName = "projects/" + projectId;
+  }
+
+  /**
+   * Creates a new secret, throwing if the secret already exists.
+   *
+   * @throws ApiException if Google API call fails.
```

Comment:
should also mention when it throws IOException

Reply:
removed IOException